File: Visitor Pattern/1. Minimal/Python/concrete_visitor_b.py
from visitor import Visitor

class ConcreteVisitorB(Visitor):
    def visit_concrete_element_a(self, element_a):
        print("ConcreteVisitorB visits ConcreteElementA:", element_a.operation_a())

    def visit_concrete_element_b(self, element_b):
        print("ConcreteVisitorB visits ConcreteElementB:", element_b.operation_b())

File: Visitor Pattern/1. Minimal/Java/ConcreteVisitorB.java
public class ConcreteVisitorB implements Visitor {
    @Override
    public void visit(ConcreteElementA element) {
        System.out.println("ConcreteVisitorB visits ConcreteElementA: " + element.operationA());
    }
  
    @Override
    public void visit(ConcreteElementB element) {
        System.out.println("ConcreteVisitorB visits ConcreteElementB: " + element.operationB());
    }
}
